src/build.rs

```diff
@@ -1,21 +1,33 @@
-//! Support to dynamic library, require in user crate a small `build.rs` script,
-//! that link to generated Rust library a filename with GHC version suffix, e.g.
-//! `libNAME-ghcVERSION.so`, `libNAME-ghcVERSION.dylib` or `NAME-ghcVERSION.dll`
-//!
-//! Version is the one of `ghc` in `$PATH`, but could be overide with
-//! `$CABAL_PACK_GHC_VERSION` env variable!
+//! Support to both static and dynamic library, require in user crate a small
+//! `build.rs` script.
 //!
 //! This build file was written with the constraint in mind of no-dependency to
 //! keep user setup simple and be easily mergeable with an existing `build.rs`
 //! automation!
 
 fn main() {
+    // Support to static library require that library name is prefixed by `C`
+    // https://gitlab.haskell.org/ghc/ghc/-/issues/22564#note_469030
     let path = format!("target/{}", std::env::var("PROFILE").unwrap());
     #[cfg(target_family = "windows")]
     let prefix = "";
     #[cfg(target_family = "unix")]
     let prefix = "lib";
     let name = env!("CARGO_PKG_NAME");
+    #[cfg(target_family = "windows")]
+    let ext = "lib";
+    #[cfg(target_family = "unix")]
+    let ext = "a";
+    let source = format!("{prefix}{name}.{ext}");
+    let target = format!("C{prefix}{name}.{ext}");
```

Comment:
the name should have the `-` changed with `_`, so we need to tweak `name`: 
```
let lib_name = name.replace("-", "_");
```

The `C` prefix that we need to do with an without is the one contained in the name. i.e. we need to end with, e.g., `libCsha_bindings.a` and `libsha_bindings.a`, so we should have another variable:

```
let lib_name_non_prefixed: &str = &lib_name[1..];
```

Reply:
Okay, so I would simply write it as `let target = format!("{prefix}C{name}.{ext}");` 😉

Reply:
wait, but if we force the user to prefix its lib name with `C`, this would result in two `C`s as a prefix, or will we not require the user anymore to prefix its lib with `C`?